### ISSUE TITLE
Docker fails to compile with "Couldn't find any versions for @grafana/data"

### ISSUE THREAD

**Issue by BloodBlight** [grafana/grafana#40537]
**What happened**:
Attempting to build either the Dockerfile or Dockerfile.ubuntu file results in the following error:
```
Step 6/40 : RUN yarn install --pure-lockfile
 ---> Running in 17baf4230e10
yarn install v1.22.5
[1/5] Validating package.json...
[2/5] Resolving packages...
error Couldn't find any versions for "@grafana/data" that matches "workspace:*"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
The command '/bin/sh -c yarn install --pure-lockfile' returned a non-zero code: 1
```
Full log here:
https://pastebin.com/bcifj6dK

**What you expected to happen**:
A docker image to be compiled.

**How to reproduce it (as minimally and precisely as possible)**:
Tried this both on my Linux Mint 20 Ulyana and a brand new clean Ubuntu Server 21.10 server (just for this build):
```
mkdir -p /home/user/Downloads
cd /home/user/Downloads
git clone https://github.com/grafana/grafana.git
cd /home/user/Downloads/grafana
mv Dockerfile Dockerfile.Other
mv Dockerfile.ubuntu Dockerfile
### NOTE: This happens on both docker files.
sudo docker build --tag 'name:LabGrafs'  .
```
Identical results on both system.

**Anything else we need to know?**:

**Environment**:
- Grafana version: Current GIT version as of Oct 15th
- Data source type & version: N/A
- OS Grafana is installed on: N/A
- User OS & Browser: Ubuntu Server 21.10 amd64
- Grafana plugins: N/A

**Comment by elliotmr** [comment]
This looks like it is related to #39082, the Dockerfile is still using Yarnv1.

**Comment by malcolmholmes** [comment]
Thx @elliotmr Looks like we need to upgrade the Yarn versions inside the two dockerfiles (`./Dockerfile` and `./Dockerfile.ubuntu`) in the root of the repo to Yarnv2.

**Comment by axelavargas** [comment]
Closed by #40672 🚀
